### PR TITLE
feature/UIDS-300 Adds indeterminate state handling to CheckboxButton

### PR DESCRIPTION
**Pull request by jasonbasuil**

### CODE REVIEW

File: src/CheckboxButton/CheckboxButton.jsx [hunk 1]
```diff
@@ -11,21 +17,31 @@ const CheckboxButton = React.forwardRef(({
...
-  />
-));
+}, ref) => {
+  useEffect(() => {
+    if (ref && ref.current) {
+      /* eslint-disable no-param-reassign */
+      ref.current.indeterminate = indeterminate;
+      /* eslint-enable no-param-reassign */
+    }
+  }, [indeterminate, ref]);
```
Comment (jasonbasuil) [inline src/CheckboxButton/CheckboxButton.jsx:27]: Not sure if I particularly like the eslint-disable line here. Any thoughts on another way around this?

File: src/index.js [hunk 2]
```diff
@@ -53,6 +53,7 @@ export {
   CardSizes,
   CheckboxButton,
   CheckboxButtonGroup,
+  CHECKED_STATES,
```
Comment (jasonbasuil) [inline src/index.js:56]: Just pulled this constant from the rails-app. Is it really helpful to export from DS though?

Comment (rroppolo) [reply src/index.js:56]: Probably overkill for just true/false and I don't know that we would necessarily need to update this for every use of CheckboxButton, but it is kind of nice to have a constant for "indeterminate" so maybe worth it just for that?

File: src/CheckboxButton/CheckboxButton.mdx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+## Stories
+
+### Default
+
+<Canvas>
+  <Story id="design-system-checkboxbutton--default" />
+</Canvas>
+
+### Indeterminate
```
Comment (rroppolo) [inline src/CheckboxButton/CheckboxButton.mdx:24]: Is it worth a sentence about what an indeterminate checkbox means? Like when a checkbox "owns" a number of sub-checkboxes and any one or more of the sub-checkboxes have a different state than the others, the owning checkbox is in the indeterminate state (or "partially" checked).

**Review by rroppolo — APPROVED** [review]
Had a few small thoughts for style/clarity but LGTM!

**Pull request merged**